### PR TITLE
pushing branch

### PR DESCRIPTION
**Pull request by almasurantor**
## Description
Working with exists with AL, just making my branch remote

### CODE REVIEW

File: src/commands/exists.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+fn check_exists(url: Url) -> Result<bool, Box<dyn std::error::Error>> {
+    let client_config = ClientConfigBuilder::default()
+        .build()?
+        .client(true);
+
+    let remote_file = RemoteFile::parse_url(url, None)?;
+    let exists = Exists::new(&remote_file);
+    let result = Exists::invoke(exists, &client_config)?;
```
Comment (dddictionary) [inline src/commands/exists.rs]: The exists command looks good! From my *limited* testing, it seems to work as intended. Might need to be tested further to detect any edge cases. Good job here though!

File: src/commands/delete.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+}
+
+fn delete_file(url: Url) -> Result<bool, Box<dyn std::error::Error>> {
+    let client_config = ClientConfigBuilder::default()
+        .build()?
+        .client(true);
+
+    let delete = Delete::new(Version::V3, url);
+    let result = delete::invoke(delete, &client_config)?;
```
Comment (dddictionary) [inline src/commands/delete.rs]: The `delete_file` function is not correct. You need to create a new `Delete` struct by providing it a remote file. the way you call the `invoke` method is also not correct. It should look like this:

```rust
fn delete_file(url: Url) -> Result<bool, Box<dyn std::error::Error>> {
    let client_config = ClientConfigBuilder::default()
        .build()?
        .client(true);
    let file = RemoteFile::parse_url(url, None).expect("Url could not be parsed");
    let delete = Delete::new(&file, None);
    Ok(delete.invoke(&client_config).is_ok())
}
```
Before copy pasting, I hope you can understand *why* exactly this works. Ask me any questions you might have!

File: src/main.rs [hunk 3]
```diff
@@ -24,5 +27,6 @@ fn main() {
     match cli.command {
         Commands::Upload(args) => commands::upload::upload_file_cmd(args),
         Commands::Download(args) => commands::download::download_file_cmd(args),
+        Commands::Exists(args) => commands::exists::exists_cmd(args),
```
Comment (dddictionary) [inline src/main.rs:33]: You have a delete command in commands/delete.rs, but you don't actually include any way for me to call it from the command line. Make sure to include an entry here so we can actually use it!

**Review by dddictionary — CHANGES_REQUESTED** [review]
Overall this looks pretty good actually. You should address and incorporate the changes I requested and these commands should be good to go. Please let me know if you ahve any questions!

**Comment by almasurantor** [comment]
-Made OwnerToken and its field public in upload.rs so it can be used in delete.rs
-Worked on delete.rs to add delete logic with cleaner messages
-Delete compiles and runs, but doesn't always fully delete — might need more debugging

File: owner_token.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (dddictionary) [inline owner_token.json:1]: add this file to .gitignore. This shouldn't be visible to people in the git repo.

File: src/commands/delete.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    // Load tokens and fetch owner token for the file
+    let tokens: HashMap<String, OwnerToken> = read_tokens_from_file("owner_token.json")?;
+    let token = tokens
+        .get(file_id)
+        .ok_or("Token not found")?
+        .owner_token
+        .clone();
+
+    let delete = Delete::new(&file, Some(token.into_bytes()));
```
Comment (dddictionary) [inline src/commands/delete.rs]: The reason why delete is not working is because you are passing the token to the wrong place. When you get the `RemoteFile` from line 32, you need to set the `owner_token` of that file to the token we read from `owner_token.json`. 

You should:
1) change the remote file to be mutable on line 32.
2) read the owner token with your already existing code.
3) use `file.set_owner_token()` to set the owner token we just read.
4) `Delete::new()` takes a reference to the file, and a `nonce`, which is an Option. Since we don't want to provide anything, it should look like `Delete::new(&file, None)` on line 43.

**Review by dddictionary — CHANGES_REQUESTED** [review]
Let me know if you have any questions. This looks pretty much finished except for the little bump in `delete.rs`

**Comment by almasurantor** [comment]
-Fixed the delete command by setting the owner token before calling delete
-Added owner_token.json to .gitignore so it doesn’t get pushed anymore

**Review by dddictionary — APPROVED** [review]
Looks really good! Nice job!
